code/utils/DiskMod.cpp

```diff
@@ -14,11 +14,16 @@ uint64_t DiskMod::GetSerializeSize() {
   // mod_type, mod_opts, and a uint64_t for the size of the serialized mod.
   uint64_t res = (2 * sizeof(uint16_t)) + sizeof(uint64_t);
 
-  if (mod_type == DiskMod::kCheckpointMod) {
+  if (mod_type == DiskMod::kCheckpointMod || mod_type == DiskMod::kSyncMod) {
     return res;
   }
 
   res += path.size() + 1;  // size() doesn't include null terminator.
+
+  if (mod_type == DiskMod::kFsyncMod) {
```

Comment:
actually, looking at this, I think we need to move it after the `res += sizeof(bool)` line cause I'm pretty sure we serialized the bool too :/